From 0fc1694f7939ec464c08951534dff71ce14d5e11 Mon Sep 17 00:00:00 2001
From: Sagar Cheluvegowda <quic_scheluve@quicinc.com>
Date: Thu, 23 Jan 2025 12:10:12 -0800
Subject: [PATCH 03/16] PENDING: net: PCS: Fix USXGMII reset sequence to poll
 for proper bit

Fix USXGMII reset sequence to poll for proper bit and modify the delay
Range between the polling to 1ms and 2ms.

Change-Id: I564f98873b1e08920ba2636edb237960f6e39139
Signed-off-by: Sagar Cheluvegowda <quic_scheluve@quicinc.com>
(cherry picked from commit 161b3b0b33ce5cb606c4561cdce19cb21434a3a5)
---
 drivers/net/pcs/pcs-xpcs-qcom.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/pcs/pcs-xpcs-qcom.c b/drivers/net/pcs/pcs-xpcs-qcom.c
index a908c928b7e7..c625a8247ecb 100644
--- a/drivers/net/pcs/pcs-xpcs-qcom.c
+++ b/drivers/net/pcs/pcs-xpcs-qcom.c
@@ -118,7 +118,7 @@ static int qcom_xpcs_poll_reset(struct dw_xpcs_qcom *qxpcs, unsigned int offset,
 	int ret;
 
 	do {
-		usleep_range(1, 20);
+		usleep_range(1000, 2000);
 		ret = qcom_xpcs_read(qxpcs, offset);
 		if (ret < 0)
 			return ret;
@@ -223,7 +223,7 @@ static int qcom_xpcs_reset_usxgmii(struct dw_xpcs_qcom *qxpcs)
 	qcom_xpcs_write(qxpcs, DW_VR_MII_PCS_DIG_CTRL1, ret | DW_USXGMII_RST);
 
 	if (!qxpcs->intr_en)
-		return qcom_xpcs_poll_reset(qxpcs, DW_VR_MII_DIG_CTRL1, SW_RST_BIT_STATUS);
+		return qcom_xpcs_poll_reset(qxpcs, DW_VR_MII_PCS_DIG_CTRL1, USXG_RST_BIT_STATUS);
 
 	return qcom_xpcs_poll_reset_usxgmii(qxpcs, DW_VR_MII_PCS_DIG_CTRL1, USXG_RST_BIT_STATUS);
 }
